### PR TITLE
KDE: update frameworks and applications

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
Some packages are still being build.

File: pkgs/development/libraries/kde-frameworks/kpackage/allow-external-paths.patch [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+--- a/src/kpackage/package.cpp
++++ b/src/kpackage/package.cpp
 @@ -820,7 +820,7 @@ PackagePrivate::PackagePrivate()
      : QSharedData(),
-       fallbackPackage(0),
-       metadata(0),
+       fallbackPackage(nullptr),
+       metadata(nullptr),
 -      externalPaths(false),
 +      externalPaths(true),
```
Comment (vandenoever) [inline pkgs/development/libraries/kde-frameworks/kpackage/allow-external-paths.patch]: Why are external paths allowed?
KPackage documentation says:
```
/**
 * @return true if paths/symlinks outside the package itself should be followed.
 * By default this is set to false for security reasons.
 */
bool allowExternalPaths() const;
```

Comment (FRidh) [inline pkgs/development/libraries/kde-frameworks/kpackage/allow-external-paths.patch]: I guess because we use symbolic links in our wrapper.

Comment (vandenoever) [inline pkgs/development/libraries/kde-frameworks/kpackage/allow-external-paths.patch]: You changed the setting, from `true` to `false` no? Did you encounter problems when the value was `false`?

Comment (FRidh) [inline pkgs/development/libraries/kde-frameworks/kpackage/allow-external-paths.patch]: That change was already present in the previous patch. Note which signs are part of the patch and which are GitHub's.

Comment (vandenoever) [inline pkgs/development/libraries/kde-frameworks/kpackage/allow-external-paths.patch]: I see now. The coloring is also applied to unchanged files. That confused me.
I've no problems with the patch as I read it, but I've not tested it.

**Comment by FRidh** [comment]
```
error: build of ‘/nix/store/bx24fh396ci9r10q7w3wxa2n8snm103z-digikam-5.4.0.drv’, ‘/nix/store/hiwvyp6d184l5i95wfawnh5hpqil5nfk-krita-3.1.1.drv’, ‘/nix/store/yln22jcj2f9x2iv3d7bbgjbl2ksxmd9k-screengrab-unstable-2016-09-12.drv’ failed
```

**Comment by FRidh** [comment]
@ttuegel I'm using this branch now without any issues. Merge?

**Comment by the-kenny** [comment]
I think we can remove `kde.digikam` (`4`) completely (instead of marking it as broken) if `digikam5` works fine.  I'm building it from this branch right now for testing.

File: pkgs/development/libraries/kde-frameworks/kpackage/allow-external-paths.patch (hunk 1 above)
Comment (ttuegel) [inline pkgs/development/libraries/kde-frameworks/kpackage/allow-external-paths.patch]: @vandenoever I added this patch. It is required for Plasma to work at all. KDE upstream considers this to be a "security feature" but that's nonsense: if your "security" requires ignoring symlinks, you don't have "security".

**Comment by ttuegel** [comment]
We can address the removal of old Digikam versions later; most KDE 4 packages will shortly be removed anyway.

**Comment by vcunat** [comment]
There are several new build failures in KDE stuff on staging, most likely due to this merge: http://hydra.nixos.org/eval/1333416?filter=&compare=1333281&full=#tabs-now-fail It often seems like they're due to using a different c++ standard, e.g. those errors about not recognizing `nullptr`.

**Comment by vcunat** [comment]
Oh, you discussed this already.  Note that it's **the new** digikam5 that's reported as failing.

**Comment by the-kenny** [comment]
Huh, I actually tried `digikam5` after merging this. I'll investigate.

**Comment by vcunat** [comment]
@the-kenny: it was only merged to staging, not master.  Now it's in master.

**Comment by the-kenny** [comment]
Oh! That explains it. I can reproduce the issue locally, will try to fix
it later today.


Vladimír Čunát <notifications@github.com> writes:

> @the-kenny: it was only merged to staging, not master.  Now it's in master.
>
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/NixOS/nixpkgs/pull/22698#issuecomment-280839351

--

**Pull request merged**